components/article_card.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+    <div v-if="articleData" class="mb-5">
+        <v-card
+            class="d-flex flex-column"
+            max-width="374"
+            min-height="500"
+        >
+            <template slot="progress">
+                <v-progress-linear
+                    color="deep-purple"
+                    height="10"
+                    indeterminate
+                ></v-progress-linear>
+            </template>
+            <v-img
+                height="250"
+                :src="articleData.urlToImage"
+            ></v-img>
+            <v-card-title @click="setSelectedArticleData" class="card-title">{{ articleData.title }}</v-card-title>
+            <v-card-text class="mb-1">
+                <div>Author: {{ articleData.author || "--" }}</div>
+                <div>Published On: {{ formattedPublishDate || "--" }}</div>
+            </v-card-text>
+            <v-spacer></v-spacer>
+            <v-divider></v-divider>
+            <v-card-actions>
+                <v-spacer></v-spacer>
+                <v-btn
+                    color="deep-purple"
+                    text
+                    @click="setSelectedArticleData"
+                >View Detail
+                </v-btn>
+                <a :href="articleData.url" target="_blank" rel="noopener noreferrer">Read Full Article</a>
+            </v-card-actions>
+        </v-card>
+    </div>
+</template>
+
+<script lang="ts">
+// Library imports
+import 'reflect-metadata'
```

Comment:
Any specific reason to use this package?
as per I know Nuxt it self has option to handle meta data .

Reply:
I am using vue-property-decorator plugin which uses above reflect-metadata internally to typecast the props
I have reffered this link: https://github.com/kaorun343/vue-property-decorator#if-youd-like-to-set-type-property-of-each-prop-value-from-its-type-definition-you-can-use-reflect-metadata.